processors/baggage/baggagetrace/processor_test.go

```diff
@@ -54,9 +51,101 @@ func TestSpanProcessorAppendsBaggageAttributes(t *testing.T) {
+	ctx := otelbaggage.ContextWithBaggage(context.Background(), baggage)
+
+	baggageKeyPredicate := func(key string) bool {
+		return strings.HasPrefix(key, "baggage.")
+	}
+
+	// create trace provider with baggage processor and test exporter
+	exporter := NewTestExporter()
+	tp := trace.NewTracerProvider(
+		trace.WithSpanProcessor(New(baggageKeyPredicate)),
+		trace.WithSpanProcessor(trace.NewSimpleSpanProcessor(exporter)),
+	)
+
+	// create tracer and start/end span
+	tracer := tp.Tracer("test")
+	_, span := tracer.Start(ctx, "test")
+	span.End()
+
+	require.Len(t, exporter.spans, 1)
+	require.Len(t, exporter.spans[0].Attributes(), 1)
+
+	want := []attribute.KeyValue{attribute.String("baggage.test", "baggage value")}
+	require.Equal(t, want, exporter.spans[0].Attributes())
+}
+
+func TestSpanProcessorAppendsBaggageAttributesWithRegexPredicate(t *testing.T) {
+	baggage, _ := otelbaggage.New()
+	baggage = addEntryToBaggage(baggage, "baggage.test", "baggage value")
+	ctx := otelbaggage.ContextWithBaggage(context.Background(), baggage)
+
+	expr := regexp.MustCompile(`^baggage\..*`)
+	baggageKeyPredicate := func(key string) bool {
+		return expr.MatchString(key)
+	}
+
+	// create trace provider with baggage processor and test exporter
+	exporter := NewTestExporter()
+	tp := trace.NewTracerProvider(
+		trace.WithSpanProcessor(New(baggageKeyPredicate)),
+		trace.WithSpanProcessor(trace.NewSimpleSpanProcessor(exporter)),
+	)
+
+	// create tracer and start/end span
+	tracer := tp.Tracer("test")
+	_, span := tracer.Start(ctx, "test")
+	span.End()
+
+	require.Len(t, exporter.spans, 1)
+	require.Len(t, exporter.spans[0].Attributes(), 1)
+
+	want := []attribute.KeyValue{attribute.String("baggage.test", "baggage value")}
+	require.Equal(t, want, exporter.spans[0].Attributes())
+}
+
+func TestOnlyAddsBaggageEntriesThatMatchPredicate(t *testing.T) {
+	baggage, _ := otelbaggage.New()
+	baggage = addEntryToBaggage(baggage, "baggage.test", "baggage value")
+	baggage = addEntryToBaggage(baggage, "foo", "bar")
+	ctx := otelbaggage.ContextWithBaggage(context.Background(), baggage)
+
+	baggageKeyPredicate := func(key string) bool {
+		return key == "baggage.test"
+	}
+
+	// create trace provider with baggage processor and test exporter
+	exporter := NewTestExporter()
+	tp := trace.NewTracerProvider(
+		trace.WithSpanProcessor(New(baggageKeyPredicate)),
+		trace.WithSpanProcessor(trace.NewSimpleSpanProcessor(exporter)),
+	)
+
+	// create tracer and start/end span
+	tracer := tp.Tracer("test")
+	_, span := tracer.Start(ctx, "test")
+	span.End()
+
+	require.Len(t, exporter.spans, 1)
+	require.Len(t, exporter.spans[0].Attributes(), 1)
+
+	want := attribute.String("baggage.test", "baggage value")
+	require.Equal(t, want, exporter.spans[0].Attributes()[0])
+}
+
+func addEntryToBaggage(baggage otelbaggage.Baggage, key, value string) otelbaggage.Baggage {
+	member, _ := otelbaggage.NewMemberRaw(key, value)
+	baggage, _ = baggage.SetMember(member)
```

Comment:
Can you ensure that there are no errors? E.g. pass `t` as first parameter, and use `require.NoError` to fail the test if there is a problem with creating+adding an entry.

Reply:
Fixed [bd60f43](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5619/commits/bd60f43ad6a414331314a5f94875cad80a8b57cd)